sklearn/decomposition/_kernel_pca.py

```diff
@@ -56,10 +63,31 @@ class KernelPCA(TransformerMixin, BaseEstimator):
         Learn the inverse transform for non-precomputed kernels.
         (i.e. learn to find the pre-image of a point)
 
-    eigen_solver : string ['auto'|'dense'|'arpack'], default='auto'
-        Select eigensolver to use. If n_components is much less than
-        the number of training samples, arpack may be more efficient
-        than the dense eigensolver.
+    eigen_solver : string {'auto', 'dense', 'arpack', 'randomized'}
+        Select eigensolver to use (default='auto'). If `n_components` is much
+        less than the number of training samples, randomized (or arpack to a
+        smaller extend) may be more efficient than the dense eigensolver.
+        Randomized SVD is performed according to the method of Halko et al.
+
+        auto :
+            the solver is selected by a default policy based on `K.shape[0]`
+            (the number of training samples) and `n_components`:
+            if the number of components to extract is lower than 80% of
```

Comment:
I wonder whether we can change the auto behaviour... Hopefully that's OK considering results should be the same. Let's see what others think

Reply:
This behaviour was copied from the one already in place in PCA: https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/decomposition/_pca.py#L161

Reply:
it's not a question of intrinsic validity, it's a question of backward compatibility. We don't want existing code to give different results just because we changed a default. Usually, we go through deprecations cycles for that. This case is particular because we're changing the 'auto' behaviour which was already the default, and in theory, the new randomized solver should give the same results. It's a bit of a grey area.

Let's see what others think.

Reply:
I am in line with your statement: yes, since the new solver should give the same results, there should probably be no deprecation cycle

Reply:
I'm okay to change the auto behaviour.